app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class TasksController < AuthenticationController
+  before_action :set_case_record
+  before_action :set_caseable
+  before_action :find_task
+  before_action :build_form, only: %i[edit update]
+
+  def show
+    respond_to do |format|
+      format.html { render template_for(:show) }
+    end
+  end
+
+  def edit
+    respond_to do |format|
+      format.html { render template_for(:edit) }
+    end
+  end
+
+  def update
+    respond_to do |format|
+      format.html do
+        if @form.update(task_params)
+          redirect_to @form.redirect_url, notice: "Task updated successfully"
+        else
+          render template_for(:edit)
+        end
+      end
+    end
+  end
+
+  private
+
+  def set_case_record
+    @case_record = @current_local_authority.case_records.find(params[:case_id])
+  end
+
+  def set_caseable
+    @enforcement = @case_record.caseable if @case_record.enforcement?
+  end
+
+  def find_task
+    slug_segments = params[:slug].to_s.split("/")
```

Comment:
Not a blocker but probably want to move this the model as it's something that should be opaque to the controller